spec/diff_calculator_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'spec_helper'
+
+include Txgh
+
+describe DiffCalculator do
+  def phrase(key, string)
+    { 'key' => key, 'string' => string }
+  end
+
+  describe '.compare' do
+    let(:diff) do
+      DiffCalculator.compare(head_phrases, diff_point_phrases)
+    end
+
+    context 'with phrases added to HEAD' do
+      let(:head_phrases) do
+        diff_point_phrases + [
+          phrase('TheNextGeneration', 'Jean Luc Picard')
+        ]
+      end
+
+      let(:diff_point_phrases) do
+        [
+          phrase('Voyager', 'Kathryn Janeway'),
+          phrase('DeepSpaceNine', 'Benjamin Sisko'),
+        ]
+      end
+
+      it 'includes the new string' do
+        expect(diff[:added].size).to eq(1)
+        expect(diff[:modified].size).to eq(0)
+        phrase = diff[:added].first
+        expect(phrase['key']).to eq('TheNextGeneration')
+        expect(phrase['string']).to eq('Jean Luc Picard')
+      end
+    end
+
+    context 'with phrases removed from HEAD' do
+      let(:head_phrases) do
+        []
+      end
+
+      let(:diff_point_phrases) do
+        [phrase('Voyager', 'Kathryn Janeway')]
+      end
+
+      it 'does not include the new string if string has been removed' do
+        expect(diff[:added].size).to eq(0)
+        expect(diff[:modified].size).to eq(0)
+      end
+    end
+
+    context 'with phrases modified in HEAD' do
+      let(:head_phrases) do
+        [phrase('TheNextGeneration', 'Jean Luc Picard (rocks)')]
+      end
+
+      let(:diff_point_phrases) do
+        [phrase('TheNextGeneration', 'Jean Luc Picard')]
+      end
+
+      it 'includes the modified string' do
+        expect(diff[:added].size).to eq(0)
+        expect(diff[:modified].size).to eq(1)
+        phrase = diff[:modified].first
+        expect(phrase['key']).to eq('TheNextGeneration')
+        expect(phrase['string']).to eq('Jean Luc Picard (rocks)')
+      end
+    end
+
+    context 'with no phrases modified, added, or removed' do
+      let(:head_phrases) do
+        [
+          phrase('TheNextGeneration', 'Jean Luc Picard'),
+          phrase('Voyager', 'Kathryn Janeway'),
```

Comment:
1. Offered amnesty to her enemies.
2. Held together a crew comprised of two opposing ideological viewpoints.
3. Treated The Doctor as a sentient being and fought for holographic rights.
4. Lost far fewer crew members than Kirk or Picard.
5. Was the first captain since Kirk to rise to the rank of admiral.
6. Brought resolution to the Q Civil War.
7. Faced the largest Borg threat with the least resources.
8. Completed a 75 year mission in 7 years.

<img src="http://4.bp.blogspot.com/-itixiBAu9Q8/Uib48xYPB_I/AAAAAAAABPc/ZrEyEQvPEDk/s1600/gzw5c.jpg" height="50%" width="50%" />